src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs

```diff
@@ -104,10 +103,21 @@ public override void TearDownOldElement(CarouselView oldElement)
 			if (ItemsView != null)
 				ItemsView.Scrolled -= CarouselViewScrolled;
 
-			ClearLayoutListener();
 			base.TearDownOldElement(oldElement);
 		}
 
+		protected override void OnAttachedToWindow()
+		{
+			base.OnAttachedToWindow();
+			AddLayoutListener();
+		}
+
+		protected override void OnDetachedFromWindow()
+		{
+			base.OnDetachedFromWindow();
+			ClearLayoutListener();
```

Comment:
Does the order matter here?

Reply:
Not sure, but just changed it anyway :-) better to have our cleanup code run first.